src/content/en/updates/2017/06/headless-karma-mocha-chai.md

```diff
@@ -0,0 +1,137 @@
+
+## Setup
+
+### Installation
+
+Install Karma, the relevant, plugins, and the test runners using **yarn**:
+
+    yarn add --dev karma karma-chrome-launcher karma-mocha karma-chai
+    yarn add --dev mocha chai
+
+or use **npm**:
+
+    npm i --save-dev karma karma-chrome-launcher karma-mocha karma-chai
+    npm i --save-dev mocha chai
+
+I'm using [Mocha](https://mochajs.org/) and [Chai](http://chaijs.com/) in this post, but
+if you're not a fan, choose your favorite assertion library that works in the browser.
+
+
+### Configure Karma
+
+Create a `karma.config.js` file that uses the `ChromeHeadless` launcher.
+
+**karma.conf.js**
+
+```javascript
+module.exports = function(config) {
+  config.set({
+    frameworks: ['mocha', 'chai'],
+    files: ['test/**/*.js'],
+    reporters: ['progress'],
+    port: 9876,  // karma web server port
+    colors: true,
+    logLevel: config.LOG_INFO,
+    browsers: ['Chrome', 'ChromeHeadless'],
+    autoWatch: false,
+    // singleRun: false, // Karma captures browsers, runs the tests and exits
+    concurrency: Infinity
+  })
+}
+```
+
+Note: Run `./node_modules/karma/bin/ init karma.conf.js` to generate the Karma configuration file.
+
+## Write a test
+
+Create a test in `/test/test.js`.
+
+**/test/test.js**
+
+```javascript
+describe('Array', () => {
+  describe('#indexOf()', () => {
+    it('should return -1 when the value is not present', () => {
+      assert.equal(-1, [1,2,3].indexOf(4));
+    });
+  });
+});
+```
+
+## Run your tests
+
+Add a `test` script in `package.json` that runs Karma with our settings.
+
+**package.json**
+
+```json
+"scripts": {
+  "test": "karma start --single-run --browsers ChromeHeadless karma.conf.js"
+}
+```
+
+When you run your tests (`yarn test`), you should Headless Chrome fire up and output
+the results to the terminal:
+
+<img width="914" alt="Output from Karma" src="https://user-images.githubusercontent.com/238208/27096498-1fd639c6-5026-11e7-9fec-58bc897d1e10.png">
+
+## Creating your own Headless Chrome launcher
+
+The `ChromeHeadless` launcher is great because it works out of the box for
+testing on Headless Chrome. It includes the appropriate Chrome flags for you and
+launches a remote debugging version of Chrome on port `9222`.
+
+However, sometimes you may want to pass custom flags to Chrome or change the
+remote debugging port the launcher uses. To do that, create a `customLaunchers`
+field that extends the base `ChromeHeadless` launcher:
+
+**karma.conf.js**
+
+```javascript
+module.exports = function(config) {
+  ...
+
+  config.set({
+    browsers: ['Chrome', 'ChromeHeadless', 'MyHeadlessChrome'],
+
+    customLaunchers: {
+      MyHeadlessChrome: {
+        base: 'ChromeHeadless',
+        flags: ['--disable-translate', '--disable-extensions', '--remote-debugging-port=9223']
+      }
+    },
+  }
+};
+```
+<br>
+
```

Comment:
good be nice to add a "next steps" field. Like personally, I'd want to know more about what type of stuff I should test, how to set up the CI